lgc/patch/PatchResourceCollect.cpp

```diff
@@ -3508,6 +3466,110 @@ void PatchResourceCollect::scalarizeGenericOutput(CallInst *call) {
   call->eraseFromParent();
 }
 
+// =====================================================================================================================
+// Clear non-specified output value in non-fragment shader stages
+void PatchResourceCollect::clearUndefinedOutput() {
+  if (m_shaderStage == ShaderStageFragment)
+    return;
+  // NOTE: If a vector or all used channels in a location are not specified, we can safely drop it and remove the output
+  // export call
+  struct CandidateInfo {
+    unsigned undefMask = 0;
+    unsigned usedMask = 0;
+    SmallVector<CallInst *> candidateCalls;
+  };
+  // Collect candidate info with undefined value at a location.
+  std::map<InOutLocationInfo, CandidateInfo> locCandidateInfoMap;
+  for (auto call : m_outputCalls) {
+    auto outputValue = call->getArgOperand(call->arg_size() - 1);
+    bool isVecOutVal = outputValue->getType()->isVectorTy();
+    bool isUndefVal = isa<UndefValue>(outputValue) || isa<PoisonValue>(outputValue);
+    unsigned index = (m_shaderStage == ShaderStageMesh || m_shaderStage == ShaderStageTessControl) ? 2 : 1;
+    bool isDynElemIndexing = !isa<ConstantInt>(call->getArgOperand(index));
+    if ((!isUndefVal && isVecOutVal) || isDynElemIndexing)
+      continue; // skip undefined vector and dynamic indexing element
```

Comment:
If we skip here, then the call is removed entirely from `m_outputCalls`. That doesn't seem right?

Reply:
Yes, you're right. We should not skip here.